packages/tangle/approver.go

```diff
@@ -0,0 +1,180 @@
+	switch len(optionalTargetObject) {
+	case 0:
+		result = &Approver{}
+	case 1:
+		result = optionalTargetObject[0]
+	default:
+		panic("too many arguments in call to ApproverFromStorageKey")
+	}
+
+	// parse the properties that are stored in the key
+	marshalUtil := marshalutil.New(key)
+	result.(*Approver).referencedMessageID, err = ParseMessageID(marshalUtil)
+	if err != nil {
+		err = fmt.Errorf("failed to parse approver referenced message ID: %w", err)
+		return
+	}
+	result.(*Approver).approverMessageID, err = ParseMessageID(marshalUtil)
+	if err != nil {
+		err = fmt.Errorf("failed to parse approver message ID: %w", err)
+		return
+	}
+	consumedBytes = marshalUtil.ReadOffset()
+
+	return
+}
+
+// ReferencedMessageID returns the ID of the message which is referenced by the approver.
+func (a *Approver) ReferencedMessageID() MessageID {
+	return a.referencedMessageID
+}
+
+// ApproverMessageID returns the ID of the message which referenced the given approved message.
+func (a *Approver) ApproverMessageID() MessageID {
+	return a.approverMessageID
+}
+
+// Bytes returns the bytes of the approver.
+func (a *Approver) Bytes() []byte {
+	return a.ObjectStorageKey()
+}
+
+// String returns the string representation of the approver.
+func (a *Approver) String() string {
+	return stringify.Struct("Approver",
+		stringify.StructField("referencedMessageID", a.ReferencedMessageID()),
+		stringify.StructField("approverMessageID", a.ApproverMessageID()),
+	)
+}
+
+// ObjectStorageKey marshals the keys of the stored approver into a byte array.
+// This includes the referencedMessageID and the approverMessageID.
+func (a *Approver) ObjectStorageKey() []byte {
+	return marshalutil.New().
+		WriteBytes(a.referencedMessageID.Bytes()).
+		WriteBytes(a.approverMessageID.Bytes()).
+		Bytes()
+}
+
+// ObjectStorageValue returns the value of the stored approver object.
+func (a *Approver) ObjectStorageValue() (result []byte) {
+	return
+}
+
+// UnmarshalObjectStorageValue unmarshals the stored bytes into an approver.
+func (a *Approver) UnmarshalObjectStorageValue(data []byte) (consumedBytes int, err error) {
+	return
+}
+
+// Update updates the approver.
+// This should should never happen and will panic if attempted.
+func (a *Approver) Update(other objectstorage.StorableObject) {
+	panic("approvers should never be overwritten and only stored once to optimize IO")
+}
+
+// interface contract (allow the compiler to check if the implementation has all of the required methods).
+var _ objectstorage.StorableObject = &Approver{}
+
+// CachedApprover is a wrapper for a stored cached object representing an approver.
+type CachedApprover struct {
+	objectstorage.CachedObject
+}
+
+// Unwrap unwraps the cached approver into the underlying approver.
+// If stored object cannot be cast into an approver or has been deleted, it returns nil.
+func (cachedApprover *CachedApprover) Unwrap() *Approver {
```

Comment:
```suggestion
func (c *CachedApprover) Unwrap() *Approver {
```
or `a` would also be fine